src/main/java/pl/cleankod/exchange/core/usecase/FindAccountAndConvertCurrencyDependsOnNullabilityOfCurrencyUseCase.java

```diff
@@ -0,0 +1,37 @@
+package pl.cleankod.exchange.core.usecase;
+
+import pl.cleankod.exchange.core.domain.Account;
+import pl.cleankod.exchange.entrypoint.model.AccountDto;
+import pl.cleankod.exchange.entrypoint.model.MoneyDto;
+
+import java.util.Currency;
+import java.util.Optional;
+
+public class FindAccountAndConvertCurrencyDependsOnNullabilityOfCurrencyUseCase {
+
+    private final FindAccountAndConvertCurrencyUseCase findAccountAndConvertCurrencyUseCase;
+    private final FindAccountUseCase findAccountUseCase;
+
+    public FindAccountAndConvertCurrencyDependsOnNullabilityOfCurrencyUseCase(FindAccountAndConvertCurrencyUseCase findAccountAndConvertCurrencyUseCase, FindAccountUseCase findAccountUseCase) {
+        this.findAccountAndConvertCurrencyUseCase = findAccountAndConvertCurrencyUseCase;
+        this.findAccountUseCase = findAccountUseCase;
+    }
+
+    public Optional<AccountDto> findAccountById(Currency currency, String id) {
+        if (currency != null) {
+            return findAccountAndConvertCurrencyUseCase.execute(Account.Id.of(id), currency).map(this::toAccountDto);
+        }
+        return findAccountUseCase.execute(Account.Id.of(id)).map(this::toAccountDto);
+    }
+
+    public Optional<AccountDto> findAccountByAccountNumber(Currency currency, Account.Number accountNumber) {
+        if (currency != null) {
```

Comment:
We could replace currency with Optional, but it would be pretty much the same. 
But instead of that I have separated methods in the controller. Previously there were 2 methods, now there are 4